examples/jsm/controls/ArcballControls.js

```diff
@@ -0,0 +1,2611 @@
+	 * @returns {Vector3} The unprojected point on the trackball plane
+	 */
+	unprojectOnTbPlane = ( camera, cursorX, cursorY, canvas, initialDistance = false ) => {
+
+		if ( camera.type == 'OrthographicCamera' ) {
+
+			this._v2_1.copy( this.getCursorPosition( cursorX, cursorY, canvas ) );
+			this._v3_1.set( this._v2_1.x, this._v2_1.y, 0 );
+
+			return this._v3_1.clone();
+
+		} else if ( camera.type == 'PerspectiveCamera' ) {
+			
+			this._v2_1.copy( this.getCursorNDC( cursorX, cursorY, canvas ) );
+
+			//unproject cursor on the near plane
+			this._v3_1.set( this._v2_1.x, this._v2_1.y, -1 );
+			this._v3_1.applyMatrix4( camera.projectionMatrixInverse );
+
+			const rayDir = this._v3_1.clone().normalize();    //unprojected ray direction
+
+			//	  camera
+			//		|\
+			//		| \
+			//		|  \
+			//	h	|	\
+			//		| 	 \
+			//		| 	  \
+			//	_ _ | _ _ _\ _ _  near plane
+			//			l
+
+			const h = this._v3_1.z;
+			const l = Math.sqrt( Math.pow( this._v3_1.x, 2 ) + Math.pow( this._v3_1.y, 2 ) );
+			let cameraGizmoDistance;
+
+			if ( initialDistance ) {
+
+				cameraGizmoDistance = this._v3_1.setFromMatrixPosition( this._cameraMatrixState0 ).distanceTo( this._v3_2.setFromMatrixPosition( this._gizmoMatrixState0 ) );
+			
+			} else {
+
+				cameraGizmoDistance = camera.position.distanceTo( this._gizmos.position );
+
+			}
+
+			/*
+			 * calculate intersection point between unprojected ray and the plane
+			 *|y = mx + q
+			 *|y = 0
+			 *
+			 * x = -q/m
+			*/
+			if ( l == 0 ) {
+
+				//ray aligned with camera
+				rayDir.set( 0, 0, 0 );
+				return rayDir;
+
+			}
+			
+			const m = h / l;
+			const q = cameraGizmoDistance;
+			const x = -q / m;
+
+			const rayLength = Math.sqrt( Math.pow( q, 2 ) + Math.pow( x, 2 ) );
+			rayDir.multiplyScalar( rayLength );
+			rayDir.z = 0;
+			return rayDir;
+
+		}
+		
+	};
+
+	/**
+	 * Update camera and gizmos state
+	 */
+	updateMatrixState = () => {
+
+		//update camera and gizmos state
+		this._cameraMatrixState.copy( this.camera.matrix );
+		this._gizmoMatrixState.copy( this._gizmos.matrix );
+
+		if ( this.camera.type == 'OrthographicCamera' ) {
+
+			this._cameraProjectionState.copy( this.camera.projectionMatrix );
+			this.camera.updateProjectionMatrix();
+			this._zoomState = this.camera.zoom;
+
+		} else if ( this.camera.type == 'PerspectiveCamera') {
+
+			this._fovState =  this.camera.fov;
+
+		}
+
+	};
+
+	/**
+	 * Update the trackball FSA
+	 * @param {STATE} newState New state of the FSA
+	 * @param {Boolean} updateMatrices If matriices state should be updated
+	 */
+	updateTbState = ( newState, updateMatrices ) => {
+
+		this._state = newState;
+		if ( updateMatrices ) {
+
+			this.updateMatrixState();
+
+		}
+
+	};
+
+	update = () => {
+
+		const EPS = 0.000001;
+
+		//check min/max parameters
+		if ( this.camera.type == 'OrthographicCamera' ) {
```

Comment:
Please use this pattern:
```js
if ( this.camera.isOrthographicCamera ) {
```
You can detect perspective camera's via `isPerspectiveCamera`.